### PR TITLE
inductor: fix size and stride comparison

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #94481

We met a case where `old.get_stride()` is a `tuple`: `(1, 16)` while `new.get_stride()` is a `list`: `[1, 16]`.
`old.get_stride() == new.get_stride()` returns `False` though they're actually equal.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/codegen/wrapper.py [hunk 1]
```diff
@@ -36,13 +36,16 @@ def make_buffer_reuse(old, new, del_func, declare, ending, as_strided):
     del_line = ""
     if old.get_name() not in V.graph.get_output_names():
         del_line = del_func(old.get_name())
-    if old.get_size() == new.get_size() and old.get_stride() == new.get_stride():
+
+    old_stride = list(old.get_stride())
+    new_stride = list(new.get_stride())
```
Comment (EikanWang) [inline torch/_inductor/codegen/wrapper.py:41]: Will `get_size` also encounter this situation? I'm wondering if we also should apply this conversion to `get_size`

Comment (chunyuan-w) [reply torch/_inductor/codegen/wrapper.py:43]: Made the same change for `get_size` as well.

Comment (jgong5) [reply torch/_inductor/codegen/wrapper.py:43]: The `get_size()` and `get_stride()` return the size and stride of the layout and the size hint shows that it should always be a list. May I know what code initializes them to a tuple?

Comment (chunyuan-w) [reply torch/_inductor/codegen/wrapper.py:43]: https://github.com/pytorch/pytorch/blob/89df0e425310c8f5210921b01a24cefc2e62e541/torch/_inductor/ir.py#L3692

`make_contiguous_strides_for` here returns a `tuple`:
https://github.com/pytorch/pytorch/blob/89df0e425310c8f5210921b01a24cefc2e62e541/torch/_prims_common/__init__.py#L1374-L1376

Below functions also return `tuple` for `stride`:
  `make_channels_last_2d_strides_for`
  `make_channels_last_3d_strides_for`
  `make_channels_last_strides_for`

**Review by jgong5 — APPROVED** [review]

**Review by EikanWang — APPROVED** [review]

File: torch/_inductor/codegen/wrapper.py (hunk 1 above)
Comment (chunyuan-w) [reply torch/_inductor/codegen/wrapper.py]: @desertfire  I've changed the `get_size` and `get_stride` functions to return `list`. Does that look good to you?

**Review by desertfire — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**